welder.py

```diff
@@ -0,0 +1,52 @@
+import subprocess, os
+from flask import (Flask, request, abort, jsonify)
+
+app = Flask(__name__)  # create the application instance
+
+
+@app.route('/api/syncTo', methods=['POST'])
+def sync_to():
+    if not request or 'destination' not in request.json:
+        return abort(400)
+    destination = request.json['destination']
+    source = os.getenv('WELDERVOLUME','')
+    if not validate_bucket(destination):
+        return jsonify({'Destination is not a valid bucket':destination}), 400
+    if (source == ''):
+        return 'Source not defined', 400
+    return run_command(source, destination)
+
+
+@app.route('/api/syncFrom', methods=['POST'])
+def sync_from():
+    if not request or 'source' not in request.json:
+        return abort(400)
+    source = request.json['source']
+    destination = os.getenv('WELDERVOLUME','')
+    if not validate_bucket(source):
+        return 'Source {0} is not a valid GCS bucket'.format(source), 400
+    if (destination == ''):
+        return 'Destination not defined', 400
+    return run_command(source, destination)
+
+
+def validate_bucket(bucket_name):
+    return bucket_name.startswith('gs://')
+
+
+def run_command(source, destination):
+    command = ['gsutil', '-m', 'rsync', '-r', source, destination]
```

Comment:
a few things that we should be doing:

we should be running rsync with the `-P` parameter so that it preserves the permissions attributes. localized files will need this in order to be rewritten by RStudio.

in that same boat, because directories are not gcs objects but will still be touched by RStudio, we need to run something like `chown -R 1000:1000 .` to assign the correct ownership to all of the directories when we are doing localization.

i wonder if the welder container will need to be run with elevated privileges in order to run `chown`...